lib/features/common/domain/entity/operation/failure.dart

```diff
@@ -0,0 +1,16 @@
+/// {@template failure.class}
+/// Failure
+/// {@endtemplate}
+base class Failure<T extends Object?> implements Exception {
```

Comment:
Why is it located in features/common/domain/entity? It looks like a universal base class. Maybe we should move it out of the features and into a conditional folder like lib/core/result? What do you think about this?